%%% introduction.tex --- 

%% Author: garamonfok@gros
%% Version: $Id: introduction.tex,v 0.0 2011/10/09 18:39:32 garamonfok Exp$

\section{What is DNA? How genes rose?}
\section{Life in DNA, from genes to repetitive elements.}
\section{Definition of neutrality}
\subsection{Neutrality in modularity}

Explanation of protein networks by to parameters probability of edge deletion $\delta$ and probability of link creation $\alpha$ after a single gene duplication \cite{Sole2008}

\section{Adaptive changes to evolutionary speed}

\section{Detection of adaptation at molecular level}

The result of selective pressures on DNA sequences are usually inferred by comparing the number of changes observed with the number of changes observed in regions known to be escaping natural selection.

In the specific context of coding regions of the genome, changes occurring at nucleotide level can be divided in two kinds depending on whether they will be reflected in translated protein sequences or not (respectively non-synonymous and synonymous changes). Even if several works outlined the footprint of natural selection in biases of synonymous changes through codon usage (see reviews \cite{Hershberg2008,Plotkin2011}), it is still assumed that its stranglehold on those silent sites is weak  \cite{Yang2008} and its usage as proxy for neutral mutation rate is used since 1980 \cite{Miyata1980}.

On the other hand, the rate of non-synonymous mutations is theoretically assumed to be subject to selective pressure as mutations occurring at those sites may have functional consequence by changing the protein sequence. Moreover its rate is significantly lower and present more variation from one gene to another when compared to the rate of silent mutations in consequence of the amount of purifying selection \cite{Kimura1985}.

Thus, assuming the proxy that silent mutations are neutral, the comparison of synonymous and non-synonymous mutation rates makes protein coding regions a perfect case in point for measuring the impact of natural selection within DNA sequences. Selective pressure can therefore be directly deduced from the ratio of non-synonymous mutation rate (dN) over synonymous mutation rate (dS), this value being associated to $\omega$:


\begin{equation} \label{eq:omega}
\omega = \frac {dN}{dS}
\end{equation}

widely used \cite{Pond2005}

\section{Grouping genes and finding evolutionary patterns}


%%% Local Variables: 
%%% mode: latex
%%% TeX-master: "../../master"
%%% End: 
